src/String_Processing/String_Hashing/String_Hashing.md

```diff
@@ -0,0 +1,42 @@
+# String Hashing
+
+We need this to compare the strings. Idea is to convert each string to integer and compare those instead of the actual strings which is O(1) operation. The conversion is done by a ***Hash-Function*** and the integer obtained corresponding to the string is called *hash* of the string.
+A widely used function is   *polynomial rolling hash function* :
+
+ ![](https://hapq.me/content/images/2019/11/Screen-Shot-2019-11-06-at-4.59.06-PM.png)
+
+where *p* and *m* are some chosen, positive numbers. *p* is a prime approximately equal to the number of characters in the input alphabet and *m* is a large number.
+Here, it is m=10^9 + 9.
+
+*The number of possible characters is higher  and pattern length can be large. So the numeric values cannot be practically stored as an integer. Therefore, the numeric value is calculated using modular arithmetic to make sure that the hash values can be stored in an integer variable.*
+
+## Implementation
+
```

Comment:
You can add syntax highlighting using triple backticks and then cpp for getting the language, that is

![image](https://user-images.githubusercontent.com/16396161/100459656-d68dd800-30eb-11eb-85e9-0a4dc0976257.png)

Change all instances of code with this format